django/middleware/common.py

```diff
@@ -161,8 +162,13 @@ def is_ignorable_request(self, request, uri, domain, referer):
         """
         Returns True if the given request *shouldn't* notify the site managers.
         """
-        # '?' in referer is identified as search engine source
+        # Ignore three specific behaviours:
+        # - empty referer
+        # - '?' in referer is identified as search engine source
+        # - referer equal to current URL as a malicious bot (#25302)
         if (not referer or
-                (not self.is_internal_request(domain, referer) and '?' in referer)):
+                (not self.is_internal_request(domain, referer) and '?' in referer) or
+                (urlparse(referer).path == uri)):
```

Comment:
This could mask actual errors if the referer is on another domain but happens to have the same path as the missing page.

I think the correct check is to compare the referer with a rebuilt URI including the scheme and domain.